test/integration/targets/find/tasks/main.yml

```diff
@@ -195,6 +196,53 @@
       that:
           - no_match_line_boundaries.matched == 0
 
+- name: When python version great than 3 and read a gbk file by utf-8
+  find:
+    paths: "{{ remote_tmp_dir_test }}"
+    patterns: "*.gbk"
+    contains: "你好世界"
+  register: fail_to_read_wrong_encoding_file
+
+- debug: var=fail_to_read_wrong_encoding_file
+
+- assert:
+      that:
+          - fail_to_read_wrong_encoding_file.msg == 'Not all paths examined, check warnings for details'
+          - >-
+              fail_to_read_wrong_encoding_file.skipped_paths[remote_tmp_dir_test] == 
+              ("Failed to read the file %s/hello_world.gbk due to an encoding error. current encoding: utf-8" % (remote_tmp_dir_test))
+  when: ansible_python_version is version('3.0', '>=')
```

Comment:
Is the test suite still running against Python 2? I think you can delete the conditionals using ansible_python_version.